src/devtools/client/debugger/src/components/Editor/LineHitCounts.tsx

```diff
@@ -167,89 +166,106 @@ function LineHitCounts({ sourceEditor }: Props) {
       // even if the mouse isn't over a valid line:
       // - The focus region just changed
       // - We refetched hit counts due to the user closing the focus bar
-      if (!isCurrentLineNumberValid && !focusRegionChanged && !hitCountMapChanged) {
+      if (!focusRegionChanged && !hitCountMapChanged) {
         return;
       }
 
       // Attempt to update just the markers for just the 100-line blocks  surrounding
       // the current line number
-      editor.eachLine(lower, upper, lineHandle => {
-        lineNumber++;
+      editor.operation(() => {
+        editor.eachLine(lower, upper, lineHandle => {
+          const currentCMLineNumber = editor.getLineNumber(lineHandle);
+          if (currentCMLineNumber === null) {
+            return;
+          }
 
-        // Skip this line if we've updated it with its current hit count already
-        if (updatedLineNumbers?.has(lineNumber)) {
-          return;
-        }
+          // CM uses 0-based indexing. Our hit counts are 1-indexed.
```

Comment:
It might be worth mentioning that our `Location` type is really where we get this from, and it's 1-indexed on lines, 0-indexed on columns. Or it might not 🙃